### PR TITLE
#3877 - Setup Program Year 2025 - 2026

### PR DESCRIPTION
**Pull request by sh16011993**
### As a part of this PR, the following are completed:

- Duplicated the BPMN files (`fulltime-assessment-2024-2025.bpmn` and `parttime-assessment-2024-2025.bpmn`) from the previous year and adjusted them as per the wiki.
- Added new 2025-2026 values in the DMN tables (for both `fulltime-assessment-decisions.dmn` and `parttime-assessment-decisions.dmn`) copying the existing 2024-2025 values.
- Copied, renamed and adjusted the following form.io definitions: `sfaa2025-26.json`, `supportingusersparent2025-2026.json`, `supportinguserspartner2025-2026.json`
- Created the new program year entry in DB
- Created a new suite of E2E tests for the new program year
- Set `is_active` to false for this program year in production has been added to the release instructions.

**Migration Revert Screenshot:**

image

--------------------------------------------------------------------------------------------------------------------------------------------

**The following tests were performed after the above mentioned updates:**

- Tested application submission and generation of NOA for the newly added program year

image

--------------------------------------------------------------------------------------------------------------------------------------------

image

### CODE REVIEW

File: sources/packages/backend/apps/db-migrations/src/sql/ProgramYear/Create-program-year-2025-2026.sql [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        partner_form_name,
+        program_year_prefix,
+        max_lifetime_bc_loan_amount
+    )
+VALUES
+    (
+        '2025-2026',
+        'Study starting between August 01, 2025 and July 31, 2026',
+        'SFAA2025-26',
+        TRUE,
```
Comment (andrepestana-aot) [inline sources/packages/backend/apps/db-migrations/src/sql/ProgramYear/Create-program-year-2025-2026.sql:19]: This should be `false`:

"Please set active to false for this program year in production until advised otherwise"

Comment (andrepestana-aot) [reply sources/packages/backend/apps/db-migrations/src/sql/ProgramYear/Create-program-year-2025-2026.sql:19]: Or should we change it just for prod?

**Review by andrepestana-aot — APPROVED** [review]
Looks good. I just leave a comment for the instructions for prod. If the migration is true, it's important to include the release instructions to change it for prod.

File: sources/packages/backend/apps/db-migrations/src/sql/ProgramYear/Create-program-year-2025-2026.sql [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        end_date,
+        parent_form_name,
+        partner_form_name,
+        program_year_prefix,
+        max_lifetime_bc_loan_amount
+    )
+VALUES
+    (
+        '2025-2026',
+        'Study starting between August 01, 2025 and July 31, 2026',
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/db-migrations/src/sql/ProgramYear/Create-program-year-2025-2026.sql:17]: Currently the program year descriptions are like this. 
image

Is there a requirement to change the description into new format? let me know if I am missing something here.

Comment (sh16011993) [reply sources/packages/backend/apps/db-migrations/src/sql/ProgramYear/Create-program-year-2025-2026.sql:17]: I just added the description (similar to the pre-existing ones as shown in your comment) for the newly added program year.

Comment (sh16011993) [reply sources/packages/backend/apps/db-migrations/src/sql/ProgramYear/Create-program-year-2025-2026.sql]: Thanks @dheepak-aot . I had made the change you are referring to but just realized that I forgot to push the file to remote.

File: sources/packages/backend/workflow/src/workflow-definitions/fulltime-assessment-decisions.dmn [hunk 3]
```diff
@@ -142,6 +142,47 @@
           <text>25</text>
         </outputEntry>
       </rule>
+      <rule id="DecisionRule_15szag4">
```
Comment (dheepak-aot) [inline sources/packages/backend/workflow/src/workflow-definitions/fulltime-assessment-decisions.dmn:145]: Not be part of the PR. 
In the FT DMNs some of them start with `2021-2022`, `2022-2023`, `2023-2024`. Is there a particular reason for that? or it is just pending a cleanup.?

image

image

image

@andrewsignori-aot @guru-aot @andrepestana-aot ?

Comment (andrewsignori-aot) [reply sources/packages/backend/workflow/src/workflow-definitions/fulltime-assessment-decisions.dmn:145]: In general, they do not need to be removed. 
If the point is that they will never going to be real production ones we can clean up (with biz approval).
For real production ones, we should keep them.
